### DISCUSSION TITLE
My first playdate game is written in rust!

### DISCUSSION THREAD

**Discussion by jcornaz** [boozook/playdate#183] [Show and tell]
Hey,

Last wekend was the [play-jam-4](https://itch.io/jam/playjam-4) and I made [this game](https://github.com/jcornaz/play-jam-4/releases/tag/play-jam-4-submission) in rust using `playdate-sys` and `cargo-playdate`!

It's very tiny, not so good and works only on device and linux simulator, But hey, I made it! Now I am impatiently waiting to receive my device to try it for real :smile:

**Comment by boozook** [comment]
Wow! 😮 Congratulations!

I'll try it on my semi-dead one-of-firsts device.

- - -

How it was?

**Reply by jcornaz** [reply]
> How it was?

The dev experience was mostly enjoyable. At least much more enjoyable than either Lua or C in my (very personal) opinion :smiley: 

For some reason, my `rust-analyzer` seems to regularly crash. Losing auto-completion and IDE error reporting was quite annoying. I don't yet understand why that's happening, but I don't think it is due to `playdate-sys`.

I went for a path where I do a lot by myself (I only use `playdate-sys`, `libm` and `serde_json`, all the rest is by me). I do that because it's fun to me. But that's a bit slow for a 72-hour game-jam. I don't regret it though, and I will continue to follow that approach, just because I find it more enjoyable :smile:.

**Reply by boozook** [reply]
@jcornaz I've tested it on my device and it works perfectly! ❤️‍🔥👍

Good level design! I hope you'll continue it and also add some sound fx.

Unfortunately I can't test crank-rotation-to-lift feel because it kinda broken on my device and need repair, so I run the game on the device controlling it with Mirror or Sim (sim can send crank events).

### Gameplay video! (short cut)

https://github.com/boozook/playdate/assets/888526/5cd37042-ce91-459b-a727-28fa879b067b
 

### Full video 

https://youtu.be/CpvRUTEd7gg

- - -

Just one little unnecessary question - why you printing "OK" for every crank change?


- - -

I've tested it on Playdate with PdOs 2.1.0-release.160721 onboard.

So you're can fix your [release](https://github.com/jcornaz/play-jam-4/releases/tag/play-jam-4-submission) description, remove this line:

> NOT TESTED ON REAL DEVICE, as I haven't received mine yet :'-(

and probably add my video if you want it ;)

**Reply by jcornaz** [reply]
> I've tested it on my device and it works perfectly! ❤️‍🔥👍

Awesome, thanks for the feedback :-)

> I can't test crank-rotation-to-lift feel because it kinda broken on my device and need reaper,

Oh shoot, that's a bummer :disappointed:.  I hope you'll manage to repair it

> Just one little unnecessary question - why you printing "OK" for every crank change?

I don't...

I don't see it, and I don't find a `println!` line in my code that would print "OK".  So, I don't know where the "OK" you see comes from.

**Reply by boozook** [reply]
Hm... 🤔 Maybe it comes from Simulator __or from OS__? It should be investigated! 🤪
[I've asked it](https://devforum.play.date/t/playdate-os-sdk-2-1-1-is-now-out/14273/8).

> I don't see it

I don't see any `"OK"` in the code too. But look at [full video with console](https://youtu.be/CpvRUTEd7gg), here it is.

**Reply by jcornaz** [reply]
Yeah, I see... strange... I don't have those when playing with the linux simulator.

**Reply by jcornaz** [reply]
btw, on the playdate dev forum, my username is `Jommag` if you want to ping me ;-)
I don't go there often though...
